launch_ros/launch_ros/actions/composable_node_container.py

```diff
@@ -51,6 +56,25 @@ def __init__(
         super().__init__(name=name, namespace=namespace, **kwargs)
         self.__composable_node_descriptions = composable_node_descriptions
 
+    @classmethod
+    def parse(cls, entity: Entity, parser: Parser):
+        """Parse node_container."""
+        _, kwargs = super().parse(entity, parser)
+
+        kwargs['package'] = entity.get_attr('pkg', data_type=str)
```

Comment:
you should use `parser.parse_substitutions()`:

```suggestion
        kwargs['package'] = parser.parse_substitutions(entity.get_attr('pkg', data_type=str))
```

Reply:
same in all the other attributes

Reply:
Exactly! Thank you, fixed in 155253e2fbee88613c2afbf84465b73061c2028c.

Question:
1. Is `parse_substitution` correct? (no **s**)
2. Is it not neccesary for `composable_nodes = entity.get_attr('composable_node', data_type=List[Entity], optional=True)`?

Reply:
> Is parse_substitution correct? (no s)

Yes, no `s`. I mistyped.

Reply:
> Is it not neccesary for composable_nodes = entity.get_attr('composable_node', data_type=List[Entity], optional=True)?

No, it's not.
You only need that where you expect to allow "embedded substitutions".

e.g. `"$(var my_config)"`